src/engine/pieces/piece.js

```diff
@@ -24,6 +24,25 @@ export default class Piece {
         return inside
     }
 
+    createMovesArray(board, possibleDir) {
+        let currentSquare = board.findPiece(this);
+        let availableMoves = []
+        possibleDir.forEach(dir => {
```

Comment:
Currently we are creating an empty array `availableMoves`, then mutating it in the forEach and then returning that array. Instead we can use the `Array.prototype.map` function that will kind of do this in one. e.g.
```
possibleDir.map(dir => {
    // map each direction to a square
    return Square.at(row, col)
})
```

We can then use this to go even further, and handle the boundary/other pieces as eg:
```
possibleDir.map(...)
    .filter(square => this.inBoundary(square.row, square.column))
    .filter(square => this.isEmptyOrCapturable(square)
```

Writing our code in this way means it is more **immutable** (ie we are not mutating an availableMoves array, but instead creating a new one)

Documentation on array.map: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map